lib/adal-angular.js

```diff
@@ -197,9 +197,35 @@
                     return false;
                 }
 
-                var routeChangeHandler = function (e, nextRoute) {
-                    var nextRouteUrl;
+                function getStates(toState) {
+                    var state = null;
+                    var states = [];
+                    if (toState.hasOwnProperty('parent')) {
+                        state = toState;
+                        while (state) {
+                            states.unshift(state);
+                            state = $injector.get('$state').get(state.parent);
```

Comment:
when state.parent is null, state = $injector.get('$state').get(state.parent) makes state= null and we exit the while loop.